### PR TITLE
Feature/convert agent parameters strings epmdj 10650

### PR DESCRIPTION
**Pull request by bodyangug**
10650

### CODE REVIEW

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt [hunk 1]
```diff
@@ -39,7 +39,7 @@ data class AgentArguments(
     val classScanDelay: Long = 0L,
     val scanClassPath: String = "",
     val packagePrefixes: String = "",
-    val coverageRetentionLimitBytes: Long = 2000L,
+    val coverageRetentionLimit: String = "2Mb",
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt:42]: I suggest changing to more lenient value - e.g. "512 mb"

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentParameters.kt [hunk 2]
```diff
@@ -29,6 +29,6 @@ data class AgentParameters(
     val isKafka: Boolean = false,
     val isCadence: Boolean = false,
     val isTlsApp: Boolean = false,
-    val coverageRetentionLimit: Long  = 2000,
+    val coverageRetentionLimit: String  = "200Mb",
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentParameters.kt:32]: Reason for defining default in multiple places.
1. There is already a default initialization in `AgentArguments.kt`
2. Decide a single place to assign defaults. Preferably in `AgentArguments.kt` for the sake of consistency

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/JvmModuleConfiguration.kt [hunk 3]
```diff
@@ -28,7 +28,7 @@ object JvmModuleConfiguration {
 
     fun getScanClassPath(): String = agentParameters.scanClassPath
 
-    fun getCoverageRetentionLimit(): Long = agentParameters.coverageRetentionLimit
+    fun getCoverageRetentionLimit(): String = agentParameters.coverageRetentionLimit
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/JvmModuleConfiguration.kt:31]: Keep return type `Long` (see comments for `DataSize.parse`)

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/jvm/JvmModuleConfigurationExternal.kt [hunk 4]
```diff
@@ -41,8 +41,8 @@ fun waitClassScanning(env: JNIEnv, thiz: jobject) =
 
 @Suppress("UNUSED")
 @CName("Java_com_epam_drill_test2code_JvmModuleConfiguration_getCoverageRetentionLimit")
-fun getCoverageRetentionLimit(env: JNIEnv, thiz: jobject): jlong =
-    callNativeLongMethod(env, thiz, JvmModuleConfiguration::getCoverageRetentionLimit)
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/jvm/JvmModuleConfigurationExternal.kt:45]: Keep `jlong` (see comments for `DataSize.parse`)

File: test2code/src/main/kotlin/com/epam/drill/test2code/JvmModuleConfiguration.kt [hunk 5]
```diff
@@ -20,5 +20,5 @@ object JvmModuleConfiguration {
     external fun getScanClassPath(): String
     external fun waitClassScanning()
     external fun getSendCoverageInterval(): Long
-    external fun getCoverageRetentionLimit(): Long
+    external fun getCoverageRetentionLimit(): String
```
Comment (RomanDavlyatshin) [inline test2code/src/main/kotlin/com/epam/drill/test2code/JvmModuleConfiguration.kt:23]: Keep return type Long (see comments for DataSize.parse)

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt [hunk 6]
```diff
@@ -99,9 +99,9 @@ data class AgentArguments(
             value = packagePrefixes,
             description = "Configure package prefixes for scanning and instrumentation",
         ),
-        AgentArguments::coverageRetentionLimitBytes.name to AgentParameter(
-            type = coverageRetentionLimitBytes.toType(),
-            value = coverageRetentionLimitBytes.toString(),
+        AgentArguments::coverageRetentionLimit.name to AgentParameter(
+            type = coverageRetentionLimit.toType(),
+            value = coverageRetentionLimit,
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt:104]: 1. Update description:
`Coverage retention queue size limit. Defaults to `*default value*`. Accepts human-readable data size format, e.g.: "512Mb", "1Gb"`

2. Does parser allow for frictional values, e.g. `"0.5 Gb"`? Check and update the description accordingly.

File: test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt [hunk 7]
```diff
@@ -30,7 +30,10 @@ interface CoverageSender {
     fun stopSendingCoverage()
 }
 
-private val COVERAGE_RETENTION_LIMIT_BYTES = BigInteger.valueOf(JvmModuleConfiguration.getCoverageRetentionLimit())
+private val COVERAGE_RETENTION_LIMIT_BYTES = DataSize.parse(JvmModuleConfiguration.getCoverageRetentionLimit())
```
Comment (RomanDavlyatshin) [inline test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt:33]: 1. No reason to use uppercase
2. Move `DataSize.parse` to `AgentArguments.kt` and just use `JvmModuleConfiguration.getCoverageRetentionLimit()` in `IntervalCoverageSender` initialization

File: test2code/src/main/kotlin/com/epam/drill/test2code/coverage/RetentionQueue.kt [hunk 8]
```diff
@@ -28,22 +28,21 @@ interface RetentionQueue {
 
 class InMemoryRetentionQueue(
     private var queue: Queue<ByteArray> = ConcurrentLinkedQueue(),
-    private val totalSizeByteLimit: BigInteger,
-    //TODO PR: check additional  Unit param. Investigate Parsing Mb to Unit.MegaBytes e.t.c
+    private val totalSizeByteLimit: BigDecimal
```
Comment (RomanDavlyatshin) [inline test2code/src/main/kotlin/com/epam/drill/test2code/coverage/RetentionQueue.kt:31]: Reason for using decimal. Number of bytes is an integer value, no need for floating point

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentParameters.kt (hunk 2 above)
Comment (RomanDavlyatshin) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentParameters.kt:32]: Edit: just set the same value `String  = "200Mb",`, leave refactoring AgentConfig/AgentArguments/AgentParams for a separate task

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/JvmModuleConfiguration.kt (hunk 3 above)
Comment (RomanDavlyatshin) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/JvmModuleConfiguration.kt:31]: obsolete

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/jvm/JvmModuleConfigurationExternal.kt (hunk 4 above)
Comment (RomanDavlyatshin) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/jvm/JvmModuleConfigurationExternal.kt:45]: obsolete

File: test2code/src/main/kotlin/com/epam/drill/test2code/JvmModuleConfiguration.kt (hunk 5 above)
Comment (RomanDavlyatshin) [reply test2code/src/main/kotlin/com/epam/drill/test2code/JvmModuleConfiguration.kt:23]: obsolete

File: test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt (hunk 7 above)
Comment (RomanDavlyatshin) [reply test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt:33]: convert to BigInteger

**Comment by RomanDavlyatshin** [comment]
MFU:
- defaults
- update description
- handle parsing errors

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentParameters.kt (hunk 2 above)
Comment (bodyangug) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentParameters.kt:32]: done

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt (hunk 6 above)
Comment (bodyangug) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt:104]: done

File: test2code/src/main/kotlin/com/epam/drill/test2code/coverage/RetentionQueue.kt (hunk 8 above)
Comment (bodyangug) [reply test2code/src/main/kotlin/com/epam/drill/test2code/coverage/RetentionQueue.kt:31]: done

File: test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt (hunk 7 above)
Comment (bodyangug) [reply test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt]: done

**Pull request merged**